pkg/controllers/repositoryserver/repositoryserver_controller_test.go

```diff
@@ -149,6 +157,186 @@ func (s *RepoServerControllerSuite) CreateGCPStorageCredentialsSecret(data map[s
 	return testutil.CreateSecret(s.kubeCli, "test-repository-server-storage-creds-", s.repoServerControllerNamespace, v1.SecretType(secrets.GCPSecretType), data)
 }
 
+func (s *RepoServerControllerSuite) TestRepositoryServerImmutability(c *C) {
+	// Create a repository server CR.
+	repoServerCR := testutil.GetTestKopiaRepositoryServerCR(s.repoServerControllerNamespace)
+	setRepositoryServerSecretsInCR(&s.repoServerSecrets, repoServerCR)
+	repoServerCRCreated, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Create(context.Background(), repoServerCR, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+	// Update the repository server CR's Immutable field.
+	repoServerCRCreated.Spec.Repository.RootPath = "/updated-test-path/"
+	_, err = s.crCli.RepositoryServers(s.repoServerControllerNamespace).Update(context.Background(), repoServerCRCreated, metav1.UpdateOptions{})
+	// Expect an error.
+	c.Assert(err, NotNil)
+	// Delete the repository server CR.
+	err = s.crCli.RepositoryServers(s.repoServerControllerNamespace).Delete(context.Background(), repoServerCRCreated.Name, metav1.DeleteOptions{})
+	c.Assert(err, IsNil)
+}
+
+func (s *RepoServerControllerSuite) TestRepositoryServerStatusIsServerReady(c *C) {
+	// Test if the repository server CR state is Ready
+	// with all the right configuration
```

Comment:
```suggestion
// TestRepositoryServerStatusIsServerReady creates a CR with correct configurations
// tests that the CR gets into created/ready state
func (s *RepoServerControllerSuite) TestRepositoryServerStatusIsServerReady(c *C) {
```

I think it's better to add comment about what test is doing as comment to test.